Documentation/Blazorise.Docs/Pages/Blog/2024-07-19_HowToStyleBlazoriseComponents/Index.md

```diff
@@ -0,0 +1,162 @@
+```html|BlazoriseInlineStylesExample
+<Alert Color="Color.Success" Visible>
+    <AlertMessage>Well done!</AlertMessage>
+    <AlertDescription Style="color:red; font-size:46px">You successfully read this important alert message.</AlertDescription>
+</Alert>
+```
+
+As we can see from [BaseComponent.customStyle](https://github.com/Megabit/Blazorise/blob/master/Source/Blazorise/Base/BaseComponent.cs#L22) all Blazorise components have support for the `Style` and `Class` and many other default attributes.
+
+### CSS classes
+
+Applying CSS classes is as straight forward as plain on Blazorise, we just specify the class like so:
+
+```html|BlazoriseCSSExample
+<Alert Color="Color.Success" Visible>
+    <AlertMessage>Well done!</AlertMessage>
+    <AlertDescription Class="bg-topo">You successfully read this important alert message.</AlertDescription>
+</Alert>
+```
+
+### Theming
+
+Blazorise has support for themes. Customize Blazorise with your theme. You can change the colors, the typography and much more using themes.
+
+To learn more about theming, head over to the [documentation page](https://blazorise.com/docs/theming)
+
+## Limitations of CSS Isolation
+
+Blazorise just like other Blazor frameworks has a limitation with CSS isolation, you see, CSS Isolation works by compiling the CSS styles and bundling them up with the Assembly.
+
+CSS isolation occurs at build time. Blazor rewrites CSS selectors to match markup rendered by the component. The rewritten CSS styles are bundled and produced as a static asset. The stylesheet is referenced inside the `<head>` tag ([location of `<head>` content](https://learn.microsoft.com/en-us/aspnet/core/blazor/project-structure?view=aspnetcore-8.0#location-of-head-and-body-content))
+
+The following `<link>` element is added by default to an app created from the Blazor project templates:
+
+```html|BlazorHtmlStylesLink
+<link href="{ASSEMBLY NAME}.styles.css" rel="stylesheet">
+```
+
+The `{ASSEMBLY NAME}` placeholder is the project's assembly name.
+
+Within the bundled file, each component is associated with a scope identifier. For each styled component, an HTML attribute is appended with the format `b-{STRING}`, where the `{STRING}` placeholder is a ten-character string generated by the framework. The identifier is unique for each app.
+
+The problem here is that, the bundle is generated at compile time of the CSS.
+
+### Let me illustrate this limitation with an example.
+
+Let's create a component in our project, call it `TestComponent`
+
+`TestComponent.razor`
+```html|TestComponentRazor
+<Div Class="isolated-class-name">
+    ...
+</Div>
+```
+
+Then create the scoped CSS file for it
+
+`TestComponent.razor.css`
+```css|TestComponentRazorCss
+.isolated-class-name {
+    /* class info here */
+}
+```
+
+This will not work, because when your application is compiled, the generated `b-{STRING}` will be different than that of Blazorise's. This is a known limitation of Blazor, but the workaround is very simple!
```

Comment:
You try to show what happens, as us as Blazor developers know what happens, because blazor will generate specific id like tags to isolate these styles, however as an unkowing reader I probably am still left not knowing exactly why.
We'd probably need to show a screenshot of the generated styles from better comprehension?

Reply:
sure, I can include the generated code and the `b-` string, and how the CSS is generated and changed. I think it would make it clear how CSS isolation works in Blazor